retrofit/src/test/java/retrofit2/CallTest.java

```diff
@@ -781,23 +782,23 @@
     Object a = new Object() {
       @Override public String toString() {
         writeCount.incrementAndGet();
-        throw new OutOfMemoryError("Broken!");
+        throw new AssertionError("Broken!");
       }
     };
     Call<String> call = service.postRequestBody(a);
 
     try {
       call.request();
       fail();
-    } catch (OutOfMemoryError e) {
+    } catch (AssertionError e) {
       assertThat(e).hasMessage("Broken!");
     }
     assertThat(writeCount.get()).isEqualTo(1);
 
     try {
       call.execute();
       fail();
-    } catch (OutOfMemoryError e) {
+    } catch (AssertionError e) {
```

Comment:
huh, just clicked that `fail()` also throws exception that is instance of `AssertionError`, exception message will catch that but still looks scary, maybe just throw `Error()` or define a custom one?

Same applies to tests below

Reply:
yeah, i thought about that but liked AssertionError because it looked like something you might actually use in real code.
plain Error seems alright, though.

Reply:
Ah, I see your reasoning, but `AssertionError` is basically a standard for testing frameworks/libraries, would be nice to avoid accidental clashing with that :)